dozer/cogs/roles.py

```diff
@@ -410,6 +474,254 @@ async def take(self, ctx, member: discord.Member, *, role: discord.Role):
+        e.add_field(name='Success!', value=f"I added created role menu [\"{name}\"]({link}) in channel {channel.mention}")
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+        await ctx.send(embed=e)
+
+    createmenu.example_usage = """
+    `{prefix}rolemenu createmenu #roles Example role menu`: Creates an empty role menu embed
+    """
+
+    @rolemenu.command(aliases=["add"])
+    @bot_has_permissions(manage_roles=True, embed_links=True)
+    @has_permissions(manage_roles=True)
+    @guild_only()
+    async def addrole(self, ctx, channel: typing.Optional[discord.TextChannel], message_id: int, role: discord.Role,
+                      emoji: typing.Union[discord.Emoji, str]):
+        """Adds a reaction role to a message or a role menu"""
+        if isinstance(emoji, discord.Emoji) and emoji.guild_id != ctx.guild.id:
+            raise BadArgument(f"The emoji {emoji} is a custom emoji not from this server!")
+
+        if role > ctx.author.top_role:
+            raise BadArgument('Cannot give roles higher than your top role!')
+
+        if role > ctx.me.top_role:
+            raise BadArgument('Cannot give roles higher than my top role!')
+
+        if role == ctx.guild.default_role:
+            raise BadArgument("Cannot give @\N{ZERO WIDTH SPACE}everyone for a level")
+
+        if role.managed:
+            raise BadArgument("I am not allowed to assign that role!")
+
+        menu_return = await RoleMenu.get_by(guild_id=ctx.guild.id, message_id=message_id)
+        menu = menu_return[0] if len(menu_return) else None
+        message = await self.safe_message_fetch(ctx, menu=menu, channel=channel, message_id=message_id)
+
+        reaction_role = ReactionRole(
+            guild_id=ctx.guild.id,
+            channel_id=message.channel.id,
+            message_id=message.id,
+            role_id=role.id,
+            reaction=str(emoji)
+        )
+
+        old_reaction = await ReactionRole.get_by(message_id=message.id, role_id=role.id)
+        if len(old_reaction):
+            await self.del_from_message(message, old_reaction[0])
+        await self.add_to_message(message, reaction_role)
+
+        if menu:
+            await self.update_role_menu(ctx, menu)
+
+        e = discord.Embed(color=blurple)
+        link = f"https://discordapp.com/channels/{ctx.guild.id}/{message.channel.id}/{message_id}"
+        shortcut = f"[{menu.name}]({link})" if menu else f"[{message_id}]({link})"
+        e.add_field(name='Success!', value=f"I added {role.mention} to message \"{shortcut}\" with reaction {emoji}")
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+        await ctx.send(embed=e)
+
+    addrole.example_usage = """
+    -----To target a role menu use this format-----
+    `{prefix}rolemenu addrole <message id> @robots 🤖`
+   -----To target a custom message use this format-----
+    `{prefix}rolemenu addrole <channel> <message id> @robots 🤖`
+    """
```

Comment:
Roles can also be provided by name or by ID. These examples should include those formats so users know they don't have to mention a giveable role (which could have quite a few members) to add it to a menu.